src/AppBundle/Controller/BlogController.php

```diff
@@ -34,12 +34,18 @@
 class BlogController extends Controller
 {
     /**
-     * @Route("/", name="blog_index")
+     * @Route("/{page}", name="blog_index", defaults={"page" = 1}, requirements={"page" : "\d+"})
```

Comment:
I think you're right. We are losing page context. In case `/blog/2` it looks like we request post with id = 2.

I like last example.Could we do that without adding additional `@Route` annotation?